bin/split_proteins_by_categories.py

```diff
@@ -73,11 +74,18 @@ def check_coordinates(self, prodigal_info, prophage_info):
         start_prophage, finish_prophage = coords.split(':')
         start_prophage = int(start_prophage)
         finish_prophage = int(finish_prophage)
-        if start_prophage <= start_protein <= finish_prophage <= finish_protein \
-            or start_protein <= start_prophage <= finish_protein <= finish_prophage \
-            or start_prophage < start_protein <= finish_protein <= finish_prophage:
-            if start_prophage <= start_protein <= finish_protein <= finish_prophage:
-                self.logger.debug(f"Protein {prodigal_info[4:20]} inside {prophage_info}")
+        prophage_length = finish_prophage - start_prophage + 1
+        # Calculate overlap
+        overlap_start = max(start_protein, start_prophage)
+        overlap_end = min(finish_prophage, finish_protein)
+        intersection = max(0, overlap_end - overlap_start + 1)
+        # Make decision about protein
+        if intersection:
+            prophage_cov = intersection / prophage_length
+            protein_cov = intersection / protein_length
+            
+            if prophage_cov > 0.9 or protein_cov > 0.9:
```

Comment:
Protein length will always be shorter that the prophage length. Evaluating the protein coverage should be enough. In the MAP we evaluate the MGE length because we can have small MGEs as short as 1 Kb, but in the case of prophages, that shouldn't be the case.